src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - You do not need to manually assign the constructor parameters to class properties if you are using TypeScript's parameter properties. You can remove the manual assignments (lines 18 and 19) as TypeScript will automatically create and assign them for `color`, `a`, `b`, and `c` since you used the `public` keyword for `a`, `b`, and `c` (lines 14, 15, 16).